sktree/datasets/hyppo.py

```diff
@@ -106,19 +99,179 @@ def make_trunk_classification(
+    and the ``band_type`` parameter. The ``band_type`` parameter controls the type of band to use, while
+    the ``rho`` parameter controls the specific scaling factor for the covariance matrix while going
+    from one dimension to the next.
+
+    For each dimension in the first distribution, there is a mean of :math:`1 / d`, where
+    ``d`` is the dimensionality. The covariance is the identity matrix.
+
+    The second distribution has a mean vector that is the negative of the first.
+    As ``d`` increases, the two distributions become closer and closer.
+    Full details for the trunk simulation can be found in :footcite:`trunk1982`.
+
+    References
+    ----------
+    .. footbibliography::
+    """
+    if n_dim < n_informative:
+        raise ValueError(
+            f"Number of informative dimensions {n_informative} must be less than number "
+            f"of dimensions, {n_dim}"
+        )
+    if simulation not in MARRON_WAND_SIMS.keys():
+        raise ValueError(
+            f"Simulation must be: trunk, trunk_overlap, trunk_mix, {MARRON_WAND_SIMS.keys()}"
+        )
+
+    rng = np.random.default_rng(seed=seed)
+
+    if rho != 0:
+        if band_type == "ma":
+            cov = _moving_avg_cov(n_informative, rho)
+        elif band_type == "ar":
+            cov = _autoregressive_cov(n_informative, rho)
+        else:
+            raise ValueError(f'Band type {band_type} must be one of "ma", or "ar".')
+    else:
+        cov = np.identity(n_informative)
+
+    # speed up computations for large multivariate normal matrix with SVD approximation
+    if n_informative > 1000:
+        mvg_sampling_method = "cholesky"
+    else:
+        mvg_sampling_method = "svd"
+
+    mixture_idx = rng.choice(
+        len(MARRON_WAND_SIMS[simulation]),  # type: ignore
+        size=n_samples // 2,
+        replace=True,
+        p=MARRON_WAND_SIMS[simulation],
+    )
+    # the parameters used for each Gaussian in the mixture for each Marron Wand simulation
+    norm_params = MarronWandSims(n_dim=n_informative, cov=cov)(simulation)
+    G = np.fromiter(
+        (
+            rng.multivariate_normal(*(norm_params[i]), size=1, method=mvg_sampling_method)
+            for i in mixture_idx
+        ),
+        dtype=np.dtype((float, n_informative)),
+    )
+
+    # as the dimensionality of the simulations increasing, we are adding more and
+    # more noise to the data using the w parameter
+    # TODO: SAMBIT I am not sure what the role of w is here.
+    w = mu_1
+    X = np.vstack(
+        (
+            rng.multivariate_normal(
+                np.zeros(n_informative), cov, n_samples // 2, method=mvg_sampling_method
+            ),
+            (1 - w)
+            * rng.multivariate_normal(
+                np.zeros(n_informative), cov, n_samples // 2, method=mvg_sampling_method
+            )
+            + w * G.reshape(n_samples // 2, n_informative),
+        )
+    )
```

Comment:
@sampan501 can you explain what the role of `w` is? I'm not sure why it's passed via `mu_1`, or if it should be?